src/Persistence/ObjectManagerDecorator.php

```diff
@@ -82,6 +87,29 @@ public function initializeObject(object $obj)
         $this->wrapped->initializeObject($obj);
     }
 
+    /** @param mixed $value */
+    public function isUninitializedObject($value): bool
+    {
+        if (! method_exists($this->wrapped, 'isUninitializedObject')) {
+            $wrappedClass = get_class($this->wrapped);
+
+            throw new BadMethodCallException(sprintf(
+                <<<'EXCEPTION'
+Context: Trying to call %s
+Problem: The wrapped ObjectManager, an instance of %s does not implement this method.
```

Comment:
`__METHOD__` does also contain the FQCN. You could shorten the "instance of" Exception text if you like.

Reply:
Not really, since the FQCN at hand is not the same as the one referred to in the "Problem" part. In the context part, I say we are calling the decorator, in the problem part, I talk about the decorated class.